File: internal/models/log_entry.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type UsageEntry struct {
	ID         int64     `json:"id"`
	ApiTokenID uuid.UUID `json:"api_token"`

	Provider string `json:"provider"`
	Latency  int    `json:"latency"`

	InputTokens  int64 `json:"input_tokens"`
	OutputTokens int64 `json:"output_tokens"`
	TotalTokens  int64 `json:"total_tokens"`
	CachedTokens int64 `json:"cached_tokens"`

	Model     string `json:"model"`
	Reasoning string `json:"reasoning"`
	Failed    bool   `json:"failed"`
	Error     string `json:"error,omitempty"`

	RequestedAt time.Time `json:"requested_at"`
	CreatedAt   time.Time `json:"created_at"`
}

File: frontend/src/lib/models/logentry.ts
import { record, text } from '../utils'

export interface LogEntry {
  provider: string
  apiKeyId: string
  latency: number
  inputTokens: number
  outputTokens: number
  totalTokens: number
  cachedTokens: number
  model: string
  reasoningEffort: string
  failed: boolean
  error: string
  requestTime: string
  createdAt: string
}
export function parseLogEntry(value: unknown): LogEntry {
  const item = record(value)
  return {
    provider: text(item.provider),
    apiKeyId: text(item.apiKeyId ?? item.api_key_id),
    latency: Number(item.latency ?? 0),
    inputTokens: Number(item.inputTokens ?? item.input_tokens ?? 0),
    outputTokens: Number(item.outputTokens ?? item.output_tokens ?? 0),
    totalTokens: Number(item.totalTokens ?? item.total_tokens ?? 0),
    cachedTokens: Number(item.cachedTokens ?? item.cached_tokens ?? 0),
    model: text(item.model),
    reasoningEffort: text(item.reasoningEffort ?? item.reasoning_effort),
    failed: item.failed === true,
    error: text(item.error),
    requestTime: text(item.requestTime ?? item.request_time),
    createdAt: text(item.createdAt ?? item.created_at),
  }
}
